### ISSUE TITLE
DASH.js in BlazorWebView errors with 'MediaSource is not supported'. Works under Windows.

### ISSUE THREAD

**Issue by warden-vt** [dotnet/maui#7909]
### Description

When using a [dash.js](https://github.com/Dash-Industry-Forum/dash.js) video player in a Blazor-maui template project I'm unable to play a video on macOS. When compiled for windows, everything functions as expected.

Windows:
<img width="1903" alt="Windows" src="https://user-images.githubusercontent.com/88406203/172928773-31b0afef-5681-4680-b56c-733fc9676a9c.png">

macOS:
<img width="1437" alt="macOS" src="https://user-images.githubusercontent.com/88406203/172928825-059fb4ca-5469-4ac8-b506-4da9cb93760d.png">

Is MediaSource supported in maccatalyst? Am I missing a plist permission? 

Attached repro project: [DASHjs - testapp.zip](https://github.com/dotnet/maui/files/8873396/DASHjs.-.testapp.zip)

I apologize if this is a known issue. If it is a known issue, is there any information regarding MediaSource support?

Thanks.

### Steps to Reproduce

Compile the application for macOS from the DASHjs - testapp.zip attached to this issue.

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

macOS

### Affected platform versions

macOS 12.3.1

### Did you find any workaround?

No

### Relevant log output

```shell
From dash.js:

Event received: error
    type: error
    (O)error:
        code: 23
        message: "mediasource is not supported"
        data: null
```

**Comment by jfversluis** [comment]
Could you maybe verify what happens if you try this in a regular Blazor app in a browser on macOS? And what if you try another video file? Both in .NET MAUI Blazor as well as the Blazor web app?

Seems to me like maybe this video format is simply not supported on Mac.

**Comment by warden-vt** [comment]
Just tested a Blazor server app. Everything works as expected on macOS.

<img width="1642" alt="Blazor Server App on macOS Safari" src="https://user-images.githubusercontent.com/88406203/173122078-d3b853d8-4238-4128-bfcd-046ace277bc9.png">


The lines particularly interesting in the dash.js video player are
`
let hasWebKit = ('WebKitMediaSource' in window);
let hasMediaSource = ('MediaSource' in window);
`

These return false when in a maui-blazer app on macOS, but they return true when in a Blazor server app visited with Safari.

**Comment by javiercn** [comment]
@warden-vt thanks for contacting us.

I'm not sure if you need an entitlement or you need to configure something on the underlying webview to make this work. Checkout this doc for [reference](https://docs.microsoft.com/en-us/aspnet/core/blazor/hybrid/?view=aspnetcore-6.0#web-view-configuration)

**Comment by warden-vt** [comment]
I haven't had much luck finding any related documentation on how I should configure the BlazozWebView on maccatalyst to allow it to use Media Source Extensions.

I did set AllowsInlineMediaPlayback to true in BlazorWebViewInitializing which seemed to be required for inline video playback on IOS. This got me thinking, is the maccatalyst [BlazorWebviewHandler.IOS.cs](https://github.com/dotnet/maui/blob/main/src/BlazorWebView/src/Maui/iOS/BlazorWebViewHandler.iOS.cs) loading an IOS WebKit web view? From what I can tell MediaSource and Media Source Extensions are not supported on IOS WebKit only MacOS and IPadOS. ([reference](https://developer.mozilla.org/en-US/docs/Web/API/MediaSource#browser_compatibility))

Safari 8+: Full Support

iOS 13+: Partial support - Exposed in Mobile Safari on iPad but not on iPhone.

Hope this helps in figuring out a direction to keep looking.

Thanks

**Comment by javiercn** [comment]
@warden-vt the MacCatalyst handler I believe uses the iOS webkit, so that might be it. 

Another thing to look with these issues is whether or not you can reproduce them with a plan old webkit window (without Blazor being involved).

**Comment by SteveSandersonMS** [comment]
Thanks @javiercn.

@warden-vt Sorry for the inconvenience here. Unfortunately it's not clear there's a defect in BlazorWebView here, as the permissions are controlled by the host application. If you're able to work out what kind of permissions are required to make this work, and after that you still feel there's a bug in BlazorWebView, please let us know and we can consider that.

**Comment by warden-vt** [comment]
@javiercn I tried a normal WebView and also set NSAppTransportSecurity to disable ATS just in case. I still get the same issue. 

As @SteveSandersonMS pointed out, I don't believe it's an issue with BlazorWebView anymore. I'm a bit unfamiliar with what permissions I should be looking for. I've looked through the WKWebViewConfiguration in BlazorWebViewInitializing to try and set some properties in the underlying WebView. I was unsuccessful in getting that to enable Media Source Extensions.

Some configuration values I tested changing in the BlazorWebViewInitializingEventArgs:

- WKPreferences.SiteSpecificQuirksModeEnabled
- WKPreferences.JavaScriptEnabled
- WKWebPagePreferences.PreferredContentMode
- WKWebPagePreferences.AllowsContentJavaScript
- WKWebViewConfiguration.AllowsInlineMediaPlayback
- WKWebViewConfiguration.RequiresUserActionForMediaPlayback

I'm at a dead end on what permissions I need to enable this. Is it possible the underlying WebView is using a WKWebKit that doesn't support Media Extensions, even though they are supported on Safari in OSX?
